javascript/app-framework/src/components/resources/FileTree/groupTables.ts

```diff
@@ -49,44 +38,35 @@ export function groupTables(tables: DataTable[]): ResourceTreeData[] {
 }
 
 function resourceNode(table: DataTable): ResourceTreeData {
+	const pathItems = (table.path as string).split('/')
+	const title =
+		pathItems[pathItems.length - 1] ?? `${table.name}.${table.format}`
 	return {
-		key: generateKey({
-			table: table.name,
-			resource: 'datasource',
-		}),
-		title: `${table.name}.${table.format}`,
+		href: `/resource/${table.name}/${title}`,
```

Comment:
This might obviate my prior comment about path structure partitioning, but it's a little hard to reason holistically about it in the PR viewer :)